back-end/controllers/QuestionController.js

```diff
@@ -8,13 +8,29 @@ const {
   editQuestion,
   removeQuestion,
   getQuestionsByUnitName,
+  getCountOfQuestionTopicsByLevel,
+  getCountOfQuestionTopicsByUnit,
 } = require('../services/QuestionService');
 
+const getCorrectFilter = (units, count) => {
+  const unitsMap = units.map((unit, index) => {
+    return { ...unit, numberOfLessons: count[index].count };
+  });
+
+  const result = unitsMap.map((item) => {
+    return { ...item._doc, numberOfLessons: item.numberOfLessons };
+  });
+
+  return result;
+};
+
 const getAllQuestions = async (req, res) => {
   try {
     const { searchUnit } = req.query;
+    const count = await getCountOfQuestionTopicsByUnit(searchUnit);
```

Comment:
You can calculate count in getQuestionsByUnitName. You don't need extra method for this